### PR TITLE
net-im/slack-bin: version bump

### PR DESCRIPTION
**Pull request by SpiderX**
1. Bump to 2.3.3
2. Dropped x86 support, because upstream does not provide a package for this arch anymore.
3. Reworked dependencies.

Gentoo-Bug: https://bugs.gentoo.org/600516

Package-Manager: portage-2.3.0

### CODE REVIEW

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	insinto /opt/${MY_PN}
+	doins -r usr/lib/${MY_PN}/*
+	fperms +x /opt/${MY_PN}/${MY_PN}
+	dosym /opt/${MY_PN}/${MY_PN} /usr/bin/${MY_PN}
+}
+
+pkg_postinst() {
+	gnome2_icon_cache_update
+}
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: also call `gnome2_icon_cache_update` in `pkg_postrm`

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	dev-libs/glib:2
+	dev-libs/atk
+	gnome-base/libgnome-keyring
+	gnome-base/gconf:2
+	sys-apps/dbus
+	net-print/cups[ssl]
+	net-misc/curl
+	virtual/udev
+	virtual/libc
+	virtual/libffi
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: please sort these alphabetically

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+"
+DEPEND="${RDEPEND}"
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: `DEPEND` not needed with binary packages

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+S="${WORKDIR}"
+
+pkg_preinst() {
+	gnome2_icon_savelist
+}
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: this needs to come after `src_install`

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+src_install() {
+	insinto /usr/share/pixmaps
+	doins usr/share/pixmaps/${MY_PN}.png
+
+	newicon -s 512 usr/share/pixmaps/${MY_PN}.png ${MY_PN}.png
+	domenu usr/share/applications/${MY_PN}.desktop
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: missing `eutils` inherit

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+	insinto /opt/${MY_PN}
+	doins -r usr/lib/${MY_PN}/*
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: use `.` instead of `*`

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+MY_PN="${PN/-bin/}"
+
+inherit unpacker gnome2-utils
+
+QA_PREBUILT="
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: `QA_*` should be `*DEPEND` and `S`

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	dev-libs/nss
+	dev-libs/glib:2
+	dev-libs/atk
+	gnome-base/libgnome-keyring
+	gnome-base/gconf:2
+	sys-apps/dbus
+	net-print/cups[ssl]
+	net-misc/curl
+	virtual/udev
+	virtual/libc
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: check what implementation of libc it links to. Very likely glibc

File: net-im/slack-bin/slack-bin-2.3.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+	media-libs/libcanberra[gtk]
+	dev-libs/nss
+	dev-libs/glib:2
+	dev-libs/atk
+	gnome-base/libgnome-keyring
+	gnome-base/gconf:2
+	sys-apps/dbus
+	net-print/cups[ssl]
+	net-misc/curl
+	virtual/udev
```
Comment (SoapGentoo) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: check the exact udev provider, this likely needs to be `virtual/libudev`

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 7 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Moved QA_ to the right place.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 2 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Sorted.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 3 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Removed.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 4 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Moved.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 5 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Added.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 6 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Thanks. Changed.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 1 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Section added.

**Comment by SpiderX** [comment]
I reworked dependencies in RDEPEND, and I am not sure that I have done everything right.
Especially, I am in doubt that I should create a dependency against mesa.
So I am asking for help.
I am providing a link to current slack's dependencies to simplify their review - http://pastebin.com/2Ayf6x2J.

File: net-im/slack-bin/slack-bin-2.3.2.ebuild (hunk 9 above)
Comment (SpiderX) [inline net-im/slack-bin/slack-bin-2.3.2.ebuild]: Forgot to mention, that dependency virtual/udev was replaced by virtual/libudev

**Comment by SpiderX** [comment]
Gentoo-Bug: https://bugs.gentoo.org/603276 should be resolved too by this PR.
Should I do something else to make this to be merged?

**Comment by Dr-Terrible** [comment]
@SpiderX an exact list of RDEPENDs can be obtained with this command:

```shell
$ qlist -e slack-bin | scanelf --needed --quiet --use-ldpath --format '%n#F' | tr ',' '\n' | sort -u | xargs qfile -S | cut -d" " -f1 | sort -u
dev-libs/atk:0
dev-libs/expat:0
dev-libs/glib:2
dev-libs/nspr:0
dev-libs/nss:0
gnome-base/gconf:2
gnome-base/libgnome-keyring:0
media-libs/alsa-lib:0
media-libs/fontconfig:1.0
media-libs/freetype:2
net-misc/curl:0
net-print/cups:0
sys-apps/dbus:0
sys-devel/gcc:5.3.0
sys-libs/glibc:2.2
x11-libs/cairo:0
x11-libs/gdk-pixbuf:2
x11-libs/gtk+:2
x11-libs/libX11:0
x11-libs/libXcomposite:0
x11-libs/libXcursor:0
x11-libs/libXdamage:0
x11-libs/libXext:0
x11-libs/libXfixes:0
x11-libs/libXi:0
x11-libs/libXrandr:0
x11-libs/libXrender:0
x11-libs/libXScrnSaver:0
x11-libs/libXtst:0
x11-libs/pango:0
```

Your ebuild is missing a few of them.

**Comment by SoapGentoo** [comment]
@SpiderX have you added the missing `RDEPEND`s?

**Comment by eliasp** [comment]
@SpiderX Upstream released `2.3.4` and `2.4.0` now, while for Linux only `2.3.4` is available yet.

**Comment by SpiderX** [comment]
@Dr-Terrible 
Thanks! Your onliner is a lifesaver :)

@SoapGentoo 
Yes.

@eliasp 
2.3.4 added in the last commit. We should not spend so much time on the next version bump, and adding 2.4.0 when it will be released, will be much quicker.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/1876a02...2992066. Thanks!

**Pull request closed without merging**